firebase/functions/src/index.ts

```diff
@@ -176,15 +175,15 @@ export const addUserToChain = functions
         await userReference.update("chainId", chainId);
         // When switching chains, you're no longer an chain-admin
         if (context.auth?.token?.role === ROLE_CHAINADMIN) {
-          await admin.auth().setCustomUserClaims(uid, { chainId: chainId });
+          await admin.auth().setCustomUserClaims(uid, {chainId: chainId});
         } else {
-          await admin
-            .auth()
-            .setCustomUserClaims(uid, {
-              chainId: chainId,
-              role: context.auth?.token?.role,
-            });
+          await admin.auth().setCustomUserClaims(uid, {
+            chainId: chainId,
+            role: context.auth?.token?.role,
+          });
         }
+
+        notifyChainAdmin(chainId, uid);
```

Comment:
This needs to be awaited

Reply:
I purposefully didn't await this because notification to the admin can be done async-ly

Reply:
I wouldn't do it like this. If you have a background job, use something like a pubsub trigger: https://firebase.google.com/docs/functions/pubsub-events

Otherwise, await everything. This can be kind-of risky. What happens if the function finished, and firebase triggers a scale-down because it goes below a threshold. Then notifyChainAdmin might not be able to finish.

I'd just await it here, notifyChainAdmin isn't that big of a function, pubsub is probably overkill.

Reply:
That's true I didn't consider the scaledown